src/test/java/gov/nih/nci/evs/api/controller/ConceptControllerTests.java

```diff
@@ -1126,6 +1128,92 @@ public void testCheckDefsAndSynsAreRight() throws Exception {
+      assertThat(assoc.getAssociation().equals("Has_Target"));
+    }
+
+    // Test with valid code -> label
+    url = baseUrl + "/ncit/associations/A7";
+    log.info("Testing url - " + url);
+    result = mvc.perform(get(url)).andExpect(status().isOk()).andReturn();
+    content = result.getResponse().getContentAsString();
+    log.info(" content = " + content);
+    resultList = new ObjectMapper().readValue(content, AssociationEntryResultList.class);
+    assertThat(resultList).isNotNull();
+    assertThat(resultList.getTimeTaken() > 0);
+    assertThat(resultList.getTotal() > 0);
+    assertThat(resultList.getParameters().getTerminology().contains("Has_Target"));
+    for (AssociationEntry assoc : resultList.getAssociationEntrys()) {
+      assertThat(assoc.getAssociation().equals("Has_Target"));
+    }
+
+    // Test with association with no data
+    url = baseUrl + "/ncit/associations/Has_CDRH_Parent";
+    log.info("Testing url - " + url);
+    result = mvc.perform(get(url)).andExpect(status().isOk()).andReturn();
+    content = result.getResponse().getContentAsString();
+    log.info(" content = " + content);
+    resultList = new ObjectMapper().readValue(content, AssociationEntryResultList.class);
+    assertThat(resultList).isNotNull();
+    assertThat(resultList.getTimeTaken() > 0);
+    assertThat(resultList.getTotal() == 0);
+    assertThat(resultList.getParameters().getTerminology().contains("Has_CDRH_Parent"));
+
+    // Test that concept subset is properly 404'd
+    url = baseUrl + "/ncit/associations/A8";
+    log.info("Testing url - " + url);
+    result = mvc.perform(get(url)).andExpect(status().isNotFound()).andReturn();
+
+    // Test pageSize
+    url = baseUrl + "/ncit/associations/Has_Target?pageSize=12";
+    log.info("Testing url - " + url);
+    result = mvc.perform(get(url)).andExpect(status().isOk()).andReturn();
+    content = result.getResponse().getContentAsString();
+    log.info(" content = " + content);
+    resultList = new ObjectMapper().readValue(content, AssociationEntryResultList.class);
+    assertThat(resultList).isNotNull();
+    assertThat(resultList.getTotal() > 0);
+    assertThat(resultList.getParameters().getTerminology().contains("12"));
+    assertThat(resultList.getAssociationEntrys().size() == 12);
+
+    // Test fromRecord
+    url = baseUrl + "/ncit/associations/Has_Target?fromRecord=1";
+    log.info("Testing url - " + url);
+    result = mvc.perform(get(url)).andExpect(status().isOk()).andReturn();
+    content = result.getResponse().getContentAsString();
+    log.info(" content = " + content);
+    resultList = new ObjectMapper().readValue(content, AssociationEntryResultList.class);
+    assertThat(resultList).isNotNull();
+    assertThat(resultList.getTotal() > 0);
+    assertThat(resultList.getParameters().getTerminology().contains("1"));
+    assertThat(resultList.getAssociationEntrys().get(0).getCode() == "C125718");
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.